### PR TITLE
first start at creating a CLI tool

### PR DESCRIPTION
**Pull request by briskt**
[IDP-544](https://itse.youtrack.cloud/issue/IDP-544)

### CODE REVIEW

File: cmd/cli/status.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+}
+
+func setupStatusCmd(rootCmd *cobra.Command) {
+	rootCmd.AddCommand(statusCmd)
+}
+
+func status() {
+	idp := viper.GetString("idp")
+	if idp == "" {
+		log.Fatalln("no IdP key is set, use --idp to set the IdP key")
```
Comment (briskt) [inline cmd/cli/status.go:31]: I will use the Cobra `MarkPersistentFlagRequired` on the next PR.

File: cmd/cli/status.go [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+	token := viper.GetString("token")
+	if token == "" {
+		log.Fatalln("no token is set, use --token to set the Terraform Cloud API token")
+	}
+	lib.SetToken(token)
+
+	workspaceName := fmt.Sprintf("idp-%s-prod-000-core", idp)
+	failoverActive, err := lib.GetWorkspaceVar(org, workspaceName, "aws_failover_active")
+	if err != nil {
+		return
```
Comment (Baggerone) [inline cmd/cli/status.go:48]: Shouldn't we say something about the error here?

**Review by Baggerone — APPROVED** [review]

File: cmd/cli/status.go (hunk 2 above)
Comment (briskt) [reply cmd/cli/status.go:48]: Yes, thanks for catching that.

File: cmd/cli/root.go [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+		// Find home directory and add ~/.config/ to the config search path
+		home, err := os.UserHomeDir()
+		cobra.CheckErr(err)
+		viper.AddConfigPath(home + "/.config")
+
+		// Add the current directory to the search path
+		viper.AddConfigPath(".")
+
+		// Set the config file name to "idp", allowing any supported file extension, e.g. idp.toml
+		viper.SetConfigName("idp")
```
[the remark this answers is not in the document]
Comment (briskt) [reply cmd/cli/root.go:72]: Sounds good

[the remark this answers is not in the document]
Comment (briskt) [reply cmd/cli/root.go:72]: I went with `idp-cli` since this tool may interact directly with AWS and not just Terraform Cloud.

**Pull request merged**